integration/oasis/stationgraph/priority_queue.go

```diff
@@ -0,0 +1,87 @@
+package stationgraph
+
+import (
+	"container/heap"
+	"math"
+
+	"github.com/golang/glog"
+)
+
+type priorityQueue struct {
```

Comment:
The `priorityQueue` and `priorityQueueImpl` seems no different. How about merge them, only keep one `priorityQueue`?

Reply:
I think they are different:
- `priorityQueue` is for external usage, the functions are `pop`, `push`
- `priorityQueueImpl` implements the interface needed by golang heap, like `Pop`, `Push`, etc

Reply:
How about `type priorityQueue *priorityQueueImpl`?